src/fsharp/Optimizer.fs

```diff
@@ -2543,6 +2543,17 @@ and TryDevirtualizeApplication cenv env (f,tyargs,args,m) =
                 MightMakeCriticalTailcall = false;
                 Info=UnknownValue})
 
+    // Analyze the name of the given symbol and rewrite AST to constant string expression with the name
+    | Expr.Val(vref,_,_),_,_ when valRefEq cenv.g vref cenv.g.nameof_vref ->
+        match PostTypeCheckSemanticChecks.extractNameOf args with
+        | Some name ->        
```

Comment:
Not even sure if it was available when I wrote that.  But tbh I think pattern matching is also ok here